app/script/message/MessageHashing.js

```diff
@@ -0,0 +1,115 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+'use strict';
+
+const Long = window.dcodeIO.Long;
+const crypto = window.crypto;
+
+window.z = window.z || {};
+window.z.message = z.message || {};
+
+z.message.MessageHashing = {
```

Comment:
How about MessageHasher for the naming?

Reply:
Let's try to expose a clean public API with just the tthreewo methods that should be used outside (namely `getAssetMessageHash`, 'getLocationMessageHash` and `getTextMessageHash`).

To do that, we could use an IIFE and create functions in there. This IIFE will then return an object with the public API